pkg/replication/replication.go

```diff
@@ -370,14 +409,18 @@ func (r *replication) ensureInReality(node types.NodeName, quitCh <-chan struct{
 
 func (r *replication) ensureHealthy(
 	node types.NodeName,
-	quitCh <-chan struct{},
+	timeoutCh <-chan struct{},
 	nodeLogger logging.Logger,
 	aggregateHealth *podHealth,
-) {
+) error {
 	for {
 		select {
-		case <-quitCh:
-			return
+		case <-r.quitCh:
+			return errQuit
+		case <-timeoutCh:
+			return errTimeout
+		case <-r.replicationCancelledCh:
```

Comment:
This function is called inside enact between the open and closed `r.enactedCh`
so if the replication gets cancelled, it will wait for `r.enactedCh` to close and so this shouldn't be a race case.

This should only close if the `renewalErrCh` gets an error in `handleRevewalErrors(...)`